datashape/coretypes.py

```diff
@@ -631,10 +629,10 @@ def shape(self):
     def itemsize(self):
         return self.ty.itemsize
 
-    def __str__(self):
-        return '?%s' % str(self.ty)
-
-    __repr__ = __str__
+    def __repr__(self):
+        return '%s(%s)' % (type(self).__name__,
```

Comment:
i'm not sure that all future `MetaType` instances will look the same modulo the prefix